woocommerce/woocommerce-android
# Fix the empty billing address issue

Fixes #2495.

| Empty billing | Empty shipping |
|---|---|
| ![image](https://user-images.githubusercontent.com/1522856/83124204-302e2f00-a0d6-11ea-95ef-7f07c8e0e1ea.png) | ![image](https://user-images.githubusercontent.com/1522856/83137167-fb77a300-a0e8-11ea-8343-2c8be3408a5d.png) | 
| **Partial billing** | **No shipping or billing** |
| ![image](https://user-images.githubusercontent.com/1522856/83123542-569f9a80-a0d5-11ea-9c89-0152284bd4ee.png) | ![image](https://user-images.githubusercontent.com/1522856/83123052-ccefcd00-a0d4-11ea-8785-e8a5837486a1.png) |
 
**To test:**

Empty billing details

1. Clear the storage before testing the app
2. Open an order with no billing address
3. Verify that the both the Billing section and `Show billing` button is hidden

Empty shipping details

1. Open an order with no shipping address
2. Verify that the `No address specified` is shown
3. Verify that the billing details are shown, if available

Partial billing details

1. Open an order with only partially filled billing details
2. Verify that the available data is shown
